base/show.jl

```diff
@@ -266,7 +273,9 @@ function show_type_name(io::IO, tn::TypeName)
     elseif globfunc
         print(io, "typeof(")
     end
-    if isdefined(tn, :module) && !(is_exported_from_stdlib(sym, tn.module) || (tn.module === Main && !hidden))
+    # Print module prefix unless type is visible from module passed to IOContext
+    # (defaulting to Main)
+    if isdefined(tn, :module) && !isvisible(sym, tn.module, get(io, :module, Main))
```

Comment:
What about pulling out `get(io, :module, Main)` and checking if it's `nothing` instead of adding the `Void` method to `isvisible`? That method seems a bit iffy to me.

Reply:
Done.